app/src/main/java/de/westnordost/streetcomplete/quests/sidewalk/AddSidewalk.kt

```diff
@@ -32,7 +32,14 @@ class AddSidewalk : OsmElementQuestType<SidewalkSides> {
           highway ~ trunk|trunk_link|primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential
           and area != yes
           and motorroad != yes
-          and !sidewalk and !sidewalk:left and !sidewalk:right and !sidewalk:both
+          and !sidewalk and !sidewalk:both
+          and (
+            (sidewalk:left and !sidewalk:right)
+            or
+            (sidewalk:right and !sidewalk:left)
+            or
+            (!sidewalk:left and !sidewalk:right)
+          )
```

Comment:
Hmm, what about other invalid combinations? Like `sidewalk` is set at the same time as `sidewalk:*`? Or `sidewalk:both` is set at the same time as `sidewalk:left`?

Rather than making the filter expression ever more complex, maybe you could use the `SidewalkParser.kt` to validate this. 

I.e., do the same as for the `AddCycleway` quest. Would also be more consistent.

Also, this approach has one flaw anyway: It treats wrong `sidewalk`-taggings like "not tagged", but that means that if there is a `highway=footway` nearby, this gets filtered out again. How it is done in `AddCycleway` takes this into account - i.e. roads with invalid, old or incomplete cycleway tagging are always asked again, independent of whether there is a cycleway nearby that might be the seperately-mapped cycleway.